routing/routing.go

```diff
@@ -154,19 +161,71 @@ func New(o Options) *Routing {
 
 	r := &Routing{log: o.Log, quit: make(chan struct{})}
 	initialMatcher, _ := newMatcher(nil, MatchingOptionsNone)
-	r.matcher.Store(initialMatcher)
+	rt := &routeTable{
+		m:       initialMatcher,
+		created: time.Now().UTC(),
+	}
+	r.routeTable.Store(rt)
 	r.startReceivingUpdates(o)
 	return r
 }
 
+// ServeHTTP ...
+func (r *Routing) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	rt := r.routeTable.Load().(*routeTable)
+	req.ParseForm()
+	createdUnix := strconv.FormatInt(rt.created.Unix(), 10)
+	ts := req.Form.Get("timestamp")
+	if ts != "" {
+		if createdUnix != ts {
+			http.Error(w, "invalid timestamp", http.StatusBadRequest)
+			return
+		}
+	}
+	offset, err := extractParam(req, "offset", 0)
+	if err != nil {
+		http.Error(w, "invalid offset", http.StatusBadRequest)
+		return
+	}
+	limit, err := extractParam(req, "limit", defaultRouteListingLimit)
+	if err != nil {
+		http.Error(w, "invalid limit", http.StatusBadRequest)
+		return
+	}
+	accept := req.Header.Get("Accept")
+	routes := slice(rt.validRoutes, offset, limit)
```

Comment:
they are not deep copy. But since the last change, they are only the eskip definitions, and here they are only being read. Do you think there can be any issues with it?